src/QuadVelocityController.cpp

```diff
@@ -462,10 +462,24 @@ double QuadVelocityController::yawFromQuaternion(
     return y;
 }
 
+void QuadVelocityController::commandForAccel(
+        const Eigen::Vector3d& accel,
+        double& pitch,
+        double& roll,
+        double& thrust)
+{
+    const auto a_hat = accel.normalized();
+    roll = -std::asin(a_hat(1));
```

Comment:
Should roll be calculated with a_hat_no_pitch similar to how a_hat_no_roll is to calculate pitch? The total thrust vector that is being projected onto the y axis shouldn't depend on the thrust in the x axis.

Reply:
No, because of the order in which they're applied you don't need it for roll

Reply:
As discussed I was mistakenly applying rotation in the global frame instead of the quad frame after each rotation.